### PR TITLE
Restored old list_daily_precipitation_totals_for_year_range function …

### PR DESCRIPTION
**Pull request by eyneill777**
…and renamed all function to use proper naming conventions

changed output of list_daily_precipitation_totals_for_year_range to be a 2d numpy array

Added a sample of folium map generation

### CODE REVIEW

File: FoliumPrecipitation.py [hunk 1]
```diff
@@ -15,11 +15,12 @@ def add_ee_layer(self, eeImageObject, visParams, name):
     control = True
   ).add_to(self)
 
-
+# Add EE drawing method to folium.
+folium.Map.add_ee_layer = add_ee_layer
```
Comment (eyneill777) [inline FoliumPrecipitation.py:19]: Moved this line out of the main function.  What this is doing is adding the add_ee_layer function as a member of the folium Map class so this is done at the same time the function is defined

File: FoliumPrecipitation.py [hunk 2]
```diff
@@ -15,11 +15,12 @@ def add_ee_layer(self, eeImageObject, visParams, name):
...
   ).add_to(self)
 
-
+# Add EE drawing method to folium.
+folium.Map.add_ee_layer = add_ee_layer
 
 
 #create a folium map and save it to the same directory as the script
-def makeMapFromImage(filename, imageOverlay, layerName, startLoc, startZoom):
+def make_map_from_image(filename, imageOverlay, layerName, startLoc, startZoom):
```
Comment (eyneill777) [inline FoliumPrecipitation.py:23]: changed naming convention

File: FoliumPrecipitation.py [hunk 3]
```diff
@@ -80,52 +101,37 @@ def viualize_data(dataset):
...
-        list.append(temp)
-    print(temp)
-
+    #Years to loop over and print data
+        
+    startYear = 2000
+    endYear = 2002
+    
+    precipVals = list_daily_precipitation_totals_for_year_range(startYear, endYear, geoArea)
+    print(precipVals)
```
Comment (eyneill777) [inline FoliumPrecipitation.py:123]: Moved above code into list_daily_precipitation_totals_for_year_range

File: FoliumPrecipitation.py [hunk 4]
```diff
@@ -80,52 +101,37 @@ def viualize_data(dataset):
...
 
-#Create overlays for the images and clip them to the area we want to analyze (uses lat/long coords)
-#precipitationOverlay = ee.Image(datalist.get(0)).clip(geoArea)
+    #Create overlays for the images and clip them to the area we want to analyze (uses lat/long coords)
+    precipitationOverlay = ee.Image(datalist.get(0)).clip(geoArea)
 
-#make map out of the overlay
-#makeMapFromImage("map", precipitationOverlay, "Precipitation", [4.1156735, -72.9301367], 5)
+    #make map out of the overlay
+    make_map_from_image("map", precipitationOverlay, "Precipitation", [4.1156735, -72.9301367], 5)
```
Comment (eyneill777) [inline FoliumPrecipitation.py:134]: changed commented out code to an example of a folium map for presentation in class

File: FoliumPrecipitation.py [hunk 5]
```diff
@@ -51,16 +52,36 @@ def getTotalPrecipitationForRegion(imlist, region, scale):
...
+        datalist = precipitation.toList(dataset.size())
+        
+        #truncate the last item if its a leap year
+        if datalist.length().getInfo() > 365:
+            datalist = datalist.remove(datalist.get(365))
+
+        #agregate the total rainfall for the area into a numpy array
+        output[year-startYear] = get_total_precipitation_for_region(datalist, region, 200)
+        
+    return output
```
Comment (eyneill777) [inline FoliumPrecipitation.py:84]: Restored this function and made changes to output the data as a 2d numpy array

**Review by eyneill777 — COMMENTED** [review]
See individual comments for granular descriptions

**Review by joshuaedwards0 — APPROVED** [review]

File: FoliumPrecipitation.py (hunk 5 above)
Comment (ReitzTimothy) [reply FoliumPrecipitation.py:84]: Lets keep this a 2D array going forward.

**Review by ReitzTimothy — APPROVED** [review]

**Pull request merged**